org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/license/LicenseUtilStatisticsTest.java

```diff
@@ -0,0 +1,161 @@
+        final Collection<Archive> archives = new ArrayList<Archive>();
+        for (DetectionStatus detectionStatus : DetectionStatus.values()) {
+            final int ordinal = detectionStatus.ordinal();
+            for (int i = 0; i < ordinal; i++) {
+                final Archive archive = new Archive(ArchiveType.JAVA, "archive" + i, "", "");
+                archive.setDetectionStatus(detectionStatus);
+                archives.add(archive);
+            }
+        }
+        final IDetectionStatusStatistics detectionStatusStatistics = LicenseUtil
+                .calculateDetectionStatusStatistics(archives);
+        Assert.assertEquals("detectionStatusStatistics (DETECTED)", 0,
+                (int) detectionStatusStatistics.get(DetectionStatus.DETECTED));
+        Assert.assertEquals("detectionStatusStatistics (MULTIPLE_DETECTED)", 1,
+                (int) detectionStatusStatistics.get(DetectionStatus.MULTIPLE_DETECTED));
+        Assert.assertEquals("detectionStatusStatistics (NOT_DETECTED)", 2,
+                (int) detectionStatusStatistics.get(DetectionStatus.NOT_DETECTED));
+        Assert.assertEquals("detectionStatusStatistics (MANUAL_DETECTED)", 3,
+                (int) detectionStatusStatistics.get(DetectionStatus.MANUAL_DETECTED));
+        Assert.assertEquals("detectionStatusStatistics (MANUAL_SELECTED)", 4,
+                (int) detectionStatusStatistics.get(DetectionStatus.MANUAL_SELECTED));
+        Assert.assertEquals("detectionStatusStatistics (MANUAL_NOT_DETECTED)", 5,
+                (int) detectionStatusStatistics.get(DetectionStatus.MANUAL_NOT_DETECTED));
+    }
+
+    /**
+     * Test for {@link LicenseUtil#calculateLegalStatusStatistics(Collection)}.
+     */
+    @Test
+    public void testCalculateLegalStatusStatisticsEmptyList() {
+        final Collection<Archive> archives = new ArrayList<Archive>();
+        final ILegalStatusStatistics legalStatusStatistics = LicenseUtil.calculateLegalStatusStatistics(archives);
+        Assert.assertEquals("legalStatusStatistics (ACCEPTED)", 0,
+                (int) legalStatusStatistics.get(LegalStatus.ACCEPTED));
+        Assert.assertEquals("legalStatusStatistics (NOT_ACCEPTED)", 0,
+                (int) legalStatusStatistics.get(LegalStatus.NOT_ACCEPTED));
+        Assert.assertEquals("legalStatusStatistics (CONFLICTING)", 0,
+                (int) legalStatusStatistics.get(LegalStatus.CONFLICTING));
+        Assert.assertEquals("legalStatusStatistics (UNKNOWN)", 0, (int) legalStatusStatistics.get(LegalStatus.UNKNOWN));
+    }
+
+    /**
+     * Test for {@link LicenseUtil#calculateLegalStatusStatistics(Collection)}.
+     */
+    @Test
+    public void testCalculateLegalStatusStatisticsRealList() {
+        final Collection<Archive> archives = new ArrayList<Archive>();
+        for (LegalStatus legalStatus : LegalStatus.values()) {
+            final int ordinal = legalStatus.ordinal();
+            for (int i = 0; i < ordinal; i++) {
```

Comment:
Same as before (to include at least one `ACCEPTED` archive):

```suggestion
            for (int i = 0; i <= ordinal; i++) {
```

NOTE: the assertions' expected-values should be updated by 1 too.

Reply:
This is intentional to test a case where there is no archive of a certain status. However, added a comment explaining this.